sentry-datasource-proxy/src/main/java/io/sentry/dsproxy/SentryQueryExecutionListener.java

```diff
@@ -0,0 +1,60 @@
+package io.sentry.dsproxy;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.Span;
+import io.sentry.SpanStatus;
+import io.sentry.util.Objects;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.stream.Collectors;
+import net.ttddyy.dsproxy.ExecutionInfo;
+import net.ttddyy.dsproxy.QueryInfo;
+import net.ttddyy.dsproxy.listener.QueryExecutionListener;
+import org.jetbrains.annotations.NotNull;
+
+/** datasource-proxy query execution listener that creates {@link Span}s around database queries. */
+@Open
+public class SentryQueryExecutionListener implements QueryExecutionListener {
+  private final @NotNull IHub hub;
+  private final @NotNull Map<String, ISpan> spans = new WeakHashMap<>();
+
+  public SentryQueryExecutionListener(final @NotNull IHub hub) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+  }
+
+  public SentryQueryExecutionListener() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public void beforeQuery(
+      final @NotNull ExecutionInfo execInfo, final @NotNull List<QueryInfo> queryInfoList) {
+    final ISpan parent = hub.getSpan();
+    if (parent != null) {
+      final ISpan span = parent.startChild("db.query", resolveSpanDescription(queryInfoList));
+      spans.put(execInfo.getConnectionId(), span);
```

Comment:
would be possible that there are multiple spans for a single connectionId?

Reply:
If they are executed in single statement or batch then I don't think we can create multiple queries. Also, query itself does not have an identifier so we would need to calculate hash from parameters to match begin and end..?

Reply:
yeah I was just wondering how this lib batch them (as its a list), think about the use case, select A and later select B, the library decides to match them within the `List<QueryInfo>` and we won't be able to capture spans for each of them but together, this would affect the average time of the span when doing select A and select B separately.
but as I don't know how the QueryInfo is batched, this is just an assumption